src/AmalgamatorSession.ts

```diff
@@ -426,6 +426,14 @@ export class AmalgamatorSession extends LoggingDebugSession {
         response.body = variables.body;
         this.sendResponse(response);
     }
+    
+    protected async evaluateRequest(response: DebugProtocol.EvaluateResponse, args: DebugProtocol.EvaluateArguments): Promise<void> {
+        const [childDap, childFrameId] = this.frameHandles.get(args.frameId ? args.frameId : 0);
```

Comment:
There is a difference between `frameId === 0` and `frameId` being undefined. I'm pretty sure what you are doing here is returning some random other frame instead of doing the evaluate in the global scope. Please [check the spec](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Evaluate)

Are you actually getting requests from VS Code with `frameId === undefined`? Is that something that you need to support?

FWIW `cdt-gdb-adapter` does [not (yet?) support the global scope evaluateRequest](https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/blob/c653a9afaecf2063501ba02d0f438eea6925f468/src/GDBDebugSession.ts#L1135) and it is probably fine to do that here too.

Reply:
Actually as there is never a `handle === 0`, the `get` will return `undefined` which will probably lead to a `TypeError: undefined is not iterable` when the return value is unpacked into `[childDap, childFrameId]`